src/settings/settings.validation.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  PipeTransform,
+  Injectable,
+  ArgumentMetadata,
+  BadRequestException,
+} from '@nestjs/common';
+import { CreateSettingDto } from './dto/create-setting.dto';
+import { UpdateSettingDto } from './dto/update-setting.dto';
+
+@Injectable()
+export class SettingsBodyValidation implements PipeTransform {
```

Comment:
Instead of defining this as a validation pipe, we can just create a custom validator class and add a custom validator for the value property in the DTO https://github.com/typestack/class-validator?tab=readme-ov-file#custom-validation-classes

Reply:
Added a custom decorator `IsSameTypeAs`